demos/http/common/src/http_demo_utils.c

```diff
@@ -67,3 +70,109 @@ int32_t connectToServerWithBackoffRetries( TransportConnect_t connectFunction,
 
     return returnStatus;
 }
+
+/*-----------------------------------------------------------*/
+
+HTTPStatus_t getUrlPath( const char * pUrl,
+                         size_t urlLen,
+                         const char ** pPath,
+                         size_t * pPathLen )
+{
+    /* http-parser status. Initialized to 0 to signify success. */
+    int parserStatus = 0;
+    struct http_parser_url urlParser;
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+
+    /* Sets all members in urlParser to 0. */
+    http_parser_url_init( &urlParser );
+
+    if( ( pUrl == NULL ) || ( pPath == NULL ) || ( pPathLen == NULL ) )
+    {
+        LogError( ( "NULL parameter passed to getUrlPath()." ) );
+        returnStatus = HTTP_INVALID_PARAMETER;
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        parserStatus = http_parser_parse_url( pUrl, urlLen, 0, &urlParser );
+
+        if( parserStatus != 0 )
+        {
+            LogError( ( "Error parsing the input URL %.*s. Error code: %d.",
+                        ( int32_t ) urlLen,
+                        pUrl,
+                        parserStatus ) );
+            returnStatus = HTTP_PARSER_INTERNAL_ERROR;
+        }
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        *pPathLen = ( size_t ) ( urlParser.field_data[ UF_PATH ].len );
+
+        if( *pPathLen == 0 )
+        {
+            returnStatus = HTTP_NO_RESPONSE;
+            *pPath = NULL;
+        }
+        else
+        {
+            *pPath = &pUrl[ urlParser.field_data[ UF_PATH ].off ];
+        }
+    }
+
+    return returnStatus;
```

Comment:
Assuming this file will follow suite to https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1318/